### PR TITLE
Joss v1.0.0

### LINKED ISSUE #756: JOSS paper for v1.0
The package has undergone substantial changes, mainly to the backend but also changes to the API. New functionality has also been added. The current JOSS paper no longer reflects the current state of the package well.

### PR DESCRIPTION
**Pull request by EdwinB12**
Closes #756

### CODE REVIEW

File: paper/paper.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    y,
+    y_transforms_list=[[], [PCATransform(n_components=16)], [PCATransform(n_components=32)]],
+)
+```
+
+The result in this case will return the best combination of model and output transformation. The Figure below shows an example result of fitting a Gaussian Process emulator in combination with PCA to a reaction-diffusion simulation (see the full [tutorial](https://alan-turing-institute.github.io/autoemulate/tutorials/emulation/02_dim_reduction.html) for a detailed overview).
+
+![GP with PCA emulator prediction for a reaction diffusion simulation compared to the ground truth.](reaction_diffusion_emulation.png)
+
+Once an emulator has been trained it can be used to generate fast predictions for new input values or to perform [downstream tasks](https://alan-turing-institute.github.io/autoemulate/tutorials/tasks/index.html) such as sensitivity analysis or model calibration. For example, to run Sobol sensitivity analysis one only needs to pass the trained emulator and some information about the data. Below is a dummy example assuming a simulation with two input parameters `param1` and `param2`, each with a specified range of values, and two outputs `output1` and `output2`:
```
Comment (radka-j) [inline paper/paper.md:133]:
```suggestion
Once an emulator has been trained it can be used to generate fast predictions for new input values or to perform [downstream tasks](https://alan-turing-institute.github.io/autoemulate/tutorials/tasks/index.html) such as sensitivity analysis or model calibration. For example, to run Sobol sensitivity analysis one only needs to pass the trained emulator and some information about the data. Below is a dummy example assuming a simulation with two input parameters `param1` and `param2`, each with a plausible range of values, and two outputs `output1` and `output2`:
```

File: paper/paper.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+# Statement of need
+
+To understand complex real-world systems, researchers and engineers often construct computer simulations. These can be computationally expensive and take minutes, hours or even days to run. A solution to this bottleneck is to approximate simulations with emulators, which can be orders of magnitudes faster [@kennedy_ohagan_2000]. Emulators are key to enabling any computationally expensive downstream tasks that require generating predictions for a high number of inputs. These include sensitivity analysis to quantify how much each input parameter affects the output, and model calibration to identify input values most likely to have generated real-world observations.
```
Comment (sgreenbury) [inline paper/paper.md:90]: Small rewording suggestion:
```suggestion
To understand complex real-world systems, researchers and engineers often construct computer simulations. These can be computationally expensive and take minutes, hours or even days to run. A solution to this bottleneck is to approximate simulations with emulators, which can be orders of magnitudes faster [@kennedy_ohagan_2000]. Emulators are key to enabling any computationally expensive downstream tasks that require generating predictions for a large number of inputs. These tasks include sensitivity analysis to quantify how much each input parameter affects the output, and model calibration to identify input values most likely to have generated real-world observations.
```

File: paper/paper.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+Emulation requires significant expertise in machine learning as well as familiarity with a broad and evolving ecosystem of tools for model training and downstream tasks. This creates a barrier to entry for domain researchers whose focus is on the underlying scientific problem. AutoEmulate [@autoemulate] lowers the barrier to entry by automating the entire emulator construction process (training, evaluation, model selection, and hyperparameter tuning). This makes emulation accessible to non-specialists while also offering a reference set of cutting-edge emulators, from classical approaches (e.g. Gaussian Processes) to modern deep learning methods, for benchmarking by experienced users.
```
Comment (sgreenbury) [inline paper/paper.md:92]: Small rewording suggestion:
```suggestion
Emulation requires significant expertise in machine learning as well as familiarity with a broad and evolving ecosystem of tools for model training and downstream tasks. This creates a barrier to entry for domain researchers whose focus is on the underlying scientific problem. AutoEmulate [@autoemulate] lowers the barrier to entry by automating the entire emulator construction process (training, evaluation, model selection, and hyperparameter tuning). This makes emulation accessible to non-specialists while also offering a reference set of cutting-edge emulators, from classical approaches (e.g. Gaussian Processes) to modern deep learning methods, enabling benchmarking for experienced users.
```

File: paper/paper.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+AutoEmulate was originally built on scikit-learn, which is well suited for traditional machine learning but less flexible for complex workflows. Version 1.0 introduces a PyTorch [@pytorch] backend that provides GPU acceleration for faster training and inference and automatic differentiation via PyTorch’s autograd system. It also makes AutoEmulate easy to integrate with other PyTorch-based tools. For example, the PyTorch refactor enables fast Bayesian model calibration using gradient-based inference methods such as Hamiltonian Monte Carlo exposed through Pyro [@pyro].
+
+AutoEmulate fills a gap in the current landscape of emulation tools as it is both accessible to newcomers and powerful enough for advanced users. It also uniquely combines emulator training with support for a wide range of downstream tasks such as sensitivity analysis, model calibration adn active learning.
```
Comment (sgreenbury) [inline paper/paper.md:98]:
```suggestion
AutoEmulate fills a gap in the current landscape of emulation tools as it is both accessible to newcomers and also flexible and powerful for advanced users. It also uniquely combines emulator training with support for a wide range of downstream tasks such as sensitivity analysis, model calibration and active learning.
```

Comment (radka-j) [reply paper/paper.md:98]: @sgreenbury I made a small tweak to your suggestions above, what do you think?

File: paper/paper.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+bc = BayesianCalibration(
+    emulator, 
+    input_parameters_ranges, 
+    observations, 
+)
+mcmc = bc.run()
+```
+
+Lastly, AutoEmulate makes it easy to integrate [custom simulators](https://alan-turing-institute.github.io/autoemulate/tutorials/simulator/01_custom_simulations.html) through subclassing and run [active learning](https://alan-turing-institute.github.io/autoemulate/tutorials/simulator/02_active_learning.html). This is particularly useful when simulations are very costly to run and one wants to select the most informative simulations to improve emulator performance at minimal computational cost.
```
[the remark this answers is not in the document]
Comment (sgreenbury) [reply paper/paper.md:174]: I think that sounds better and also like the addition of simulator-in-the-loop.

File: paper/paper.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+problem = {
+    'num_vars': 2,
+    'names': ["param1", "param2"],  
+    'bounds': input_parameters_ranges.values(),
+    'output_names': ["output1", "output2"],
+}
+
+sa = SensitivityAnalysis(emulator, problem=problem)
+sobol_df = sa.run()
+```
```
Comment (sgreenbury) [inline paper/paper.md:157]: It might be worth linking directly to a more applied example with a sentence something like this after the code? Though this level of detail may be unncessary given the link is already included above for downstream tasks.
```suggestion
```
A more complete application to a cardiovascular simulator is demonstrated [here](https://alan-turing-institute.github.io/autoemulate/tutorials/tasks/01_emulation_sensitivity.html).
```

File: paper/paper.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
```
Comment (sgreenbury) [inline paper/paper.md:158]: It might be worth linking directly to a more applied example with a sentence something like this after the code? Though this level of detail may be unncessary given the link is already included above for downstream tasks.
```suggestion
A more complete application of sensitivity analysis to a cardiovascular simulator is demonstrated [here](https://alan-turing-institute.github.io/autoemulate/tutorials/tasks/01_emulation_sensitivity.html).
```

File: paper/paper.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+observations = {'output1': 0.5, 'output2': 7.2}
+
+bc = BayesianCalibration(
+    emulator, 
+    input_parameters_ranges, 
+    observations, 
+)
+mcmc = bc.run()
+```
+
```
Comment (sgreenbury) [inline paper/paper.md:173]: It might be worth linking directly to a more applied example with a sentence something like this after the code? As above though, this level of detail may be unncessary given the link is already included above for downstream tasks.
```suggestion
A more complete application of Bayesian model calibration to simulating an epidemic is demonstrated [here](https://alan-turing-institute.github.io/autoemulate/tutorials/tasks/03_bayes_calibration.html).
```

File: paper/paper.md [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+    'names': ["param1", "param2"],  
+    'bounds': input_parameters_ranges.values(),
+    'output_names': ["output1", "output2"],
+}
+
+sa = SensitivityAnalysis(emulator, problem=problem)
+sobol_df = sa.run()
+```
+A more complete application of sensitivity analysis to a cardiovascular simulator is demonstrated [here](https://alan-turing-institute.github.io/autoemulate/tutorials/tasks/01_emulation_sensitivity.html).
+The PyTorch backend enables fast Bayesian model calibration using gradient-based inference methods such as Hamiltonian Monte Carlo with Pyro. AutoEmulate provides a simple interface for this given a trained PyTorch emulator, input parameter ranges (same as in the sensitivity analysis example), and real-world observations:
```
Comment (radka-j) [inline paper/paper.md:159]:
```suggestion

The PyTorch backend enables fast Bayesian model calibration using gradient-based inference methods such as Hamiltonian Monte Carlo with Pyro. AutoEmulate provides a simple interface for this given a trained PyTorch emulator, input parameter ranges (same as in the sensitivity analysis example), and real-world observations:
```

File: paper/paper.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+      orcid: 0009-0004-1124-469X
+      affiliation: 2
+    - name: Marjan Famili
+      orcid: 
+      affiliation: 1
+    - name: Chris Sprague
+      orcid: 
+      affiliation: 1
+    - name: Paolo Conti
+      orcid:
```
Comment (ContiPaolo) [inline paper/paper.md:37]: My orcID is 0000-0003-4062-2560

File: paper/paper.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+# Statement of need
+
+To understand complex real-world systems, researchers and engineers often construct computer simulations. These can be computationally expensive and take minutes, hours or even days to run. A solution to this bottleneck is to approximate simulations with emulators, which can be orders of magnitudes faster [@kennedy_ohagan_2000]. Emulators are key to enabling any computationally expensive downstream tasks that require generating predictions for a large number of inputs. These tasks include sensitivity analysis to quantify how much each input parameter affects the output, and model calibration to identify input values most likely to have generated real-world observations.
```
Comment (ContiPaolo) [inline paper/paper.md:90]: We never mention that the main characteristics of emulator is having uncertainty quantification.
I would add " ... [@Kennedy_ohagan_2000] and provide uncertainty quantification on the estimates."
Just to mention it at least once

File: paper/paper.md [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+
+ae = AutoEmulate(
+    x, 
+    y,
+    y_transforms_list=[
+      [],
+      [PCATransform(n_components=16)],
+      [PCATransform(n_components=32)]
+    ],
+)
```
Comment (ContiPaolo) [inline paper/paper.md:131]: Since we are showing an image with results with PCA + GP, I think it wouldn't hurt to add `models=["GaussianProcess"]`, to make it more clear.

```
  ae = AutoEmulate(
      x, 
      y,
      y_transforms_list=[
        [],
        [PCATransform(n_components=16)],
        [PCATransform(n_components=32)]
      ],
      models=["GaussianProcess"],
  )
```

File: paper/paper.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+
+# Statement of need
+
+To understand complex real-world systems, researchers and engineers often construct computer simulations. These can be computationally expensive and take minutes, hours or even days to run. A solution to this bottleneck is to approximate simulations with emulators, which can be orders of magnitudes faster [@kennedy_ohagan_2000]. Emulators are key to enabling any computationally expensive downstream tasks that require generating predictions for a large number of inputs. These tasks include sensitivity analysis to quantify how much each input parameter affects the output, and model calibration to identify input values most likely to have generated real-world observations.
```
Comment (marjanfamili) [inline paper/paper.md:90]:
```suggestion
Complex physical systems are often modelled using computer simulations. Depending on the complexity of the system, these simulation can be computationally expensive and time-consuming. This bottleneck can be resolved by approximating simulations with emulators, which can be orders of magnitudes faster [@kennedy_ohagan_2000]. Emulators are key to enabling any computationally expensive downstream tasks that require generating predictions for a large number of inputs. These tasks include sensitivity analysis to quantify the impact of each input parameter on the output as well as model calibration to identify input values most likely to have generated real-world observations.
```

File: paper/paper.md [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+
+AutoEmulate v1.0 introduces easy-to-use interfaces for common emulation tasks. By providing these within a single package, the package enables users to construct sequential workflows. For example, sensitivity analysis can be applied to narrow the parameter space to a few key variables before calibrating this reduced set against real world data.  AutoEmulate also supports direct integration of custom simulators and active learning, in which the tool adaptively selects informative simulations to run to improve emulator performance at minimal computational cost.
```
Comment (marjanfamili) [inline paper/paper.md:94]:
```suggestion
AutoEmulate v1.0 introduces easy-to-use interfaces for common emulation tasks. By providing these tasks within a single package it enables users to construct sequential workflows. For instance, sensitivity analysis can be applied in order to narrow down the parameter space to the key variables. This allows the user to calibrate the much smaller reduced set to match the output of the model to real-world observations.  AutoEmulate also supports direct integration of custom simulators and active learning, in which the tool adaptively selects informative simulations to run to improve emulator performance at minimal computational cost.
```

**Review by marjanfamili — APPROVED** [review]
real-world is sometimes `real-world` sometimes `real world`

I have made some suggestions where it was harder to read

File: paper/paper.md (hunk 11 above)
Comment (radka-j) [reply paper/paper.md]: I added a suggestion along those lines below.

File: paper/paper.md [hunk 15]
```diff
@@ -0,0 +1,182 @@
...
+      affiliation: 1
+      equal-contrib: true
+    - name: Ed Chalstrey
+      orcid: 0000-0003-2560-1294
+      affiliation: 1
+    - name: Edwin Brown
+      orcid: 0009-0004-1124-469X
+      affiliation: 2
+    - name: Marjan Famili
+      orcid: 0009-0003-0426-3721
```
Comment (radka-j) [inline paper/paper.md:31]:
```suggestion
```

File: paper/paper.md [hunk 16]
```diff
@@ -0,0 +1,181 @@
...
+      affiliation: 1
+      equal-contrib: true
+    - name: Ed Chalstrey
+      orcid: 0000-0003-2560-1294
+      affiliation: 1
+    - name: Edwin Brown
+      orcid: 0009-0004-1124-469X
+      affiliation: 2
+    - name: Marjan Famili
+      affiliation: 1
```
Comment (radka-j) [inline paper/paper.md:31]:
```suggestion
      orcid: 0009-0003-0426-3721
      affiliation: 1
```
